waltz-tools/src/main/java/com/wepay/waltz/tools/cluster/ClusterCli.java

```diff
@@ -289,20 +302,147 @@ private void failMissingValidationResults(List<PartitionValidationResults> parti
+                if (zkPartitionToStorageNodeMap.get(id).size() != partitionToStorageStatusMap.get(id).size()) {
+                    partitionAssignmentValidationResult = new ValidationResult(
+                        ValidationResult.ValidationType.PARTITION_ASSIGNMENT_ZK_STORAGE_CONSISTENCY,
+                        ValidationResult.Status.FAILURE, "Error: ZooKeeper and StorageNode Partition Assignment "
+                        + "mismatch");
+                } else {
+                    partitionAssignmentValidationResult = new ValidationResult(
+                        ValidationResult.ValidationType.PARTITION_ASSIGNMENT_ZK_STORAGE_CONSISTENCY,
+                        ValidationResult.Status.SUCCESS, "");
+                }
+                partitionValidationResults.validationResultsMap.put(partitionAssignmentValidationResult.type,
+                    partitionAssignmentValidationResult);
+
+                // Verify quorum
+                List<String> quorumList = new ArrayList<>();
+                Map<String, Boolean> storageReadWriteStatusMap = partitionToStorageStatusMap.get(id);
+                storageReadWriteStatusMap.forEach((storageConnectString, status) -> {
+                    if (status) {
+                        quorumList.add(storageConnectString);
+                    }
+                });
+
+                if (quorumList.size() <= (storageConnections.size() / 2)) {
+                    quorumValidationResult = new ValidationResult(
+                        ValidationResult.ValidationType.PARTITION_QUORUM_STATUS, ValidationResult.Status.FAILURE,
+                        "Error: Quorum is not achieved");
+                } else {
+                    quorumValidationResult =
+                        new ValidationResult(ValidationResult.ValidationType.PARTITION_QUORUM_STATUS,
+                            ValidationResult.Status.SUCCESS, "");
+                }
+                partitionValidationResults.validationResultsMap.put(quorumValidationResult.type,
+                    quorumValidationResult);
+            }
+        }
+
+        private Map<Integer, Boolean> buildPartitionStatusPerStorage(String storageConnection,
+                                                    Integer storageAdminPort,
+                                                    SslContext sslContext,
+                                                    StoreMetadata storeMetadata) throws Exception {
+            StoreParams storeParams = storeMetadata.getStoreParams();
+            StorageAdminClient storageAdminClient = null;
+
+            try {
+                String[] storageHostAndPortArray = storageConnection.split(":");
+                String storageHost = storageHostAndPortArray[0];
+
+                storageAdminClient = new StorageAdminClient(storageHost, storageAdminPort, sslContext,
+                    storeParams.key, storeParams.numPartitions);
+                storageAdminClient.open();
+
```

Comment:
It looks like that timeout is per call to each storage node. Does it mean the maximum total waiting time for the caller method = Timeout  x  number of storage names ?
Have the 5 second timeout been chosen with knowledge of the above?

Reply:
hmm i think you are right, the total waiting time might be `Timeout x number of storage nodes`. I missed out this point. good catch. Will fix it.
I don't think timeout 5sec was chosen with this in mind.

Reply:
Fixed.